frontend/src/app/auth.service.ts

```diff
@@ -36,6 +36,16 @@ export class AuthService {
       });
   }
 
+  logout() {
+      this.http.get(this._logoutUrl, { withCredentials: true })
+        .subscribe(rsp => {
```

Comment:
This is not a good pattern to subscribe within this method. I would suggest returning the observable (with i.e. current user), like `logout(): Observable<User>`.

The login within your subscribe could happen in ie. `map` rxjs function.

So, this function could look like:
```
return this.http.get(...).map(response => { // mind the return
  if (this._currentUser) {
    this._currentUser = undefined;
    this.changeUserEvent.emit(this._currentUser);
}
```

Then, anywhere this method will be supposed to be called, it will happen with `this.authService.logout().subscribe()`.

It lets you to respond somehow to the logout result, so the call could look like:
```
this.authService.logout().subscribe(user => {
  ...do sth, print "farewell, dear user X" or redirect, whatever
}
```